boards/arm/96b_neonkey/doc/96b_neonkey.rst

```diff
@@ -0,0 +1,194 @@
+- 15 General purpose LEDs
+- GPIO with external interrupt capability
+- I2C (3)
+- SPI (1)
+- I2S (1)
+
+Supported Features
+==================
+
+The Zephyr 96b_neonkey board configuration supports the following hardware
+features:
+
++-----------+------------+-------------------------------------+
+| Interface | Controller | Driver/Component                    |
++===========+============+=====================================+
+| NVIC      | on-chip    | nested vector interrupt controller  |
++-----------+------------+-------------------------------------+
+| SYSTICK   | on-chip    | system clock                        |
++-----------+------------+-------------------------------------+
+| UART      | on-chip    | serial port                         |
++-----------+------------+-------------------------------------+
+| GPIO      | on-chip    | gpio                                |
++-----------+------------+-------------------------------------+
+| PINMUX    | on-chip    | pinmux                              |
++-----------+------------+-------------------------------------+
+| FLASH     | on-chip    | flash                               |
++-----------+------------+-------------------------------------+
+| SPI       | on-chip    | spi                                 |
++-----------+------------+-------------------------------------+
+| I2C       | on-chip    | i2c                                 |
++-----------+------------+-------------------------------------+
+
+More details about the board can be found at `96Boards website`_.
+
+The default board configuration can be found in the defconfig file:
+
+        ``boards/arm/96b_neonkey/96b_neonkey_defconfig``
+
+Connections and IOs
+===================
+
+LED
+---
+
+- LED1 / User1 LED = PB12
+- LED2 / User2 LED = PB13
+- LED3 / User3 LED = PB14
+- LED4 / User4 LED = PB15
+
+Push buttons
+------------
+
+- BUTTON = RST (SW1)
+- BUTTON = USR (SW2)
+
+System Clock
+============
+
+96Boards Neonkey could be driven by internal oscillator as well as main PLL clock.
+By default System clock is sourced by PLL clock at 100MHz, driven by internal oscillator.
+
+Serial Port
+===========
+
+On 96Boards Neonkey Zephyr console output is assigned to USART1.
+Default settings are 115200 8N1.
+
+I2C
+---
+
+96Boards Neonkey board has up to 3 I2Cs. The default I2C mapping for Zephyr is:
+
+- I2C1_SCL : PB6
+- I2C1_SDA : PB7
+- I2C2_SCL : PB10
+- I2C2_SDA : PB3
+- I2C3_SCL : PA8
+- I2C3_SCL : PB4
+
+Programming and Debugging
+*************************
+
+Building
+========
+
+Build the Zephyr kernel and application with,
+
+.. code-block:: console
+
+   $ cd <zephyr_root_path>
+   $ source zephyr-env.sh
+   $ cd $ZEPHYR_BASE/samples/hello_world/
+   $ make BOARD=96b_neonkey
+
+Flashing
+========
+
+96Boards Neonkey can be flashed by two methods, one using the ROM
+bootloader and another using the SWD debug port (which requires additional
+hardware). Flashing using the ROM bootloader requires a special activation
```

Comment:
Is this enough information for folks to know how to flash using the ROM boot loader?  No examples are given here for this process.

Reply:
@Mani-Sadhasivam, looking at the schematic, it seems that SW2 has to be pressed (so BOOT1=0), and BOOT0 has to be pulled up to VDD (e.g. wire pins 1 and 5 on the JTAG connector together) before power is applied -- is that right?

It might also be worth adding a reference to www.st.com/resource/en/application_note/cd00167594.pdf, which describes this in full detail

Reply:
@mbolivar yes, you are right! I will add that information. Thanks for spotting this.

Reply:
@mbolivar @dbkinder  Should I also add information about how to use the flash utility for ROM bootloader?

Reply:
That would make me happy.